### DISCUSSION TITLE
Autoload Scenes properly typed for parser

### DISCUSSION THREAD

**Discussion by FoxyFox909** [godotengine/godot-proposals#10524] [Engine Core]
When using Scene autoloads (picked `.tscn` file for autoload),  and you have `unsafe_property_access` warnings turned on, the debugger will be angry with you and tell you that the property is not present on the inferred type "Node".

This warning goes away when you make the autoload be a script instead (picked `.gd` file for autoload).

This makes it cumbersome to use Autoload scenes if you need to access any of the properties in its the script of its root node (which does work, but triggers warnings).

`AutoloadScript` is a `.gd` autoload. `AutoloadScene` is a `.tscn` autoload with the same script as `AutoloadScript` set on its root Node.
![image](https://github.com/user-attachments/assets/97d3f63d-c42e-439e-af81-1bd3c8f9f39b)

Here is an MRP: [autoload-mrp.zip](https://github.com/user-attachments/files/16703727/autoload-mrp.zip)

**Comment by jupiterbjy** [comment]
I think you forgot to discard some files!

Here's MRE discarding `.godot` and `.git*` (2.13MiB -> 2.98KiB): [autoload-mrp.zip](https://github.com/user-attachments/files/16712061/autoload-mrp.zip)

---

I cannot reproduce on windows ( *v4.3.stable.official.77dcf97d8* )

![image](https://github.com/user-attachments/assets/5eee5904-aeb4-410f-a6a8-5453b7efb7ea)

<br>

Warning flag is up

![image](https://github.com/user-attachments/assets/1f79b23f-3227-43a2-ab65-095af288634f)

<br>

---

If issue is persistent, wonder if adding `class_name` helps as it now no longer becomes mere `Node` type where `Node.a` doesn't exist

```gdscript
class_name SomeClassName extends Node

var a : int = 12345

func _ready() -> void:
	print("Autoload autoloaded: ", name)
```

**Comment by FoxyFox909** [comment]
I attempted adding `class_name`, but that also didn't help on my end. It's like the parser only considers the autolaoded scene to be the return type of `Node.instantiate()`
